client/src/components/related_items/Modal.jsx

```diff
@@ -1,26 +1,31 @@
 import React, {useState, useRef, useEffect} from 'react';
 import {faXmark} from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { isDOMComponent } from 'react-dom/test-utils';
 
 
 const Modal = ({showModal, closeModal, children})=>{
 
     const modalRef = useRef(null);
 
     const clickedOutside = (e) => {
-
+        if(showModal && modalRef.current && !modalRef.current.contains(e.target)) {
+            closeModal();
+        }
     }
     
-    
     useEffect(()=>{
-
+        document.addEventListener('mousedown', clickedOutside);
```

Comment:
There might be a way to write this without this type of function, I certainly used it a few times but I think theres more intuitive ways to write it. But like i said when it was in my code, if it ain't broke don't fix it

Reply:
looks good.